prime-dotnet-webapi/Services/EnrolleePaperSubmissionService.cs

```diff
@@ -264,5 +264,130 @@ public async Task FinalizeSubmissionAsync(int enrolleeId)
+                .Where(
+                    pe => pe.GPID.StartsWith(PaperGpidPrefix)
+                    && pe.Id == paperEnrolmentId
+                    && _context.EnrolleeLinkedEnrolment.Any(
+                        link => link.PaperEnrolleeId == pe.Id
+                        )
+                )
+                .AnyAsync();
+
+            if (paperEnrollee)
+            {
+                return false;
+            }
+
+            var enrolleeLinkedEnrolment = await _context.EnrolleeLinkedEnrolment
+                .Where(ele => ele.EnrolleeId == enrolmentId)
+                .SingleOrDefaultAsync();
+
+            enrolleeLinkedEnrolment.PaperEnrolleeId = paperEnrolmentId;
+            enrolleeLinkedEnrolment.EnrolmentLinkDate = DateTime.Now;
+            enrolleeLinkedEnrolment.Confirmed = isConfirmed;
+
+            await _context.SaveChangesAsync();
+
+            return true;
+        }
+
+        public async Task CreateInitialLinkAsync(int enrolleeId, string userProvidedGpid)
+        {
+            var enrolleeLinkedEnrolment = await _context.EnrolleeLinkedEnrolment
+                .Where(ele => ele.EnrolleeId == enrolleeId)
+                .SingleOrDefaultAsync();
+
+            if (enrolleeLinkedEnrolment == null)
+            {
+                var newLinkedEnrolment = new EnrolleeLinkedEnrolment
+                {
+                    EnrolleeId = enrolleeId,
+                    UserProvidedGpid = userProvidedGpid,
+                };
+
+                _context.Add(newLinkedEnrolment);
+            }
+            else
+            {
+                enrolleeLinkedEnrolment.UserProvidedGpid = userProvidedGpid;
+            }
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task UpdateLinkedGpidAsync(int enrolleeId, string userUpdatedGpid)
+        {
+            var enrolleeLinkedEnrolment = await _context.EnrolleeLinkedEnrolment
+                .Where(ele => ele.EnrolleeId == enrolleeId)
+                .SingleOrDefaultAsync();
+
+            enrolleeLinkedEnrolment.UserProvidedGpid = userUpdatedGpid;
+            _context.SaveChanges();
+        }
+
+        public async Task<string> GetLinkedGpidAsync(int enrolleeId)
+        {
+            var value = await _context.EnrolleeLinkedEnrolment
+            .AsNoTracking()
+            .Where(ele => ele.EnrolleeId == enrolleeId)
+            .Select(ele => new
+            {
+                ele.UserProvidedGpid
+            })
+            .DecompileAsync()
+            .SingleOrDefaultAsync();
+
+            return value.UserProvidedGpid;
+        }
+        public async Task<PermissionsRecord> GetPermissionsRecordAsync(int enrolleeId)
+        {
+            return await _context.Enrollees
+                .AsNoTracking()
+                .Where(e => e.Id == enrolleeId)
+                .Select(e => new PermissionsRecord { UserId = e.UserId })
+                .SingleOrDefaultAsync();
+        }
```

Comment:
Why have you duplicated this method?

Reply:
I duplicated that as we don't have a reference to `EnrolleeService` class in `EnrolleePaperSubmissionController` to call that method I could either, create an object of `EnrolleeService`, or create a reference to `EnrolleeService`. An alternative can be to just call `EnrolleeService.GetPermissionRecordAsync` inside the one in `EnrolleePaperSubmissionService`. Let me know if you have a preference for which way you want to use.

Reply:
Fixed, added reference to `EnrolleeService` to `EnrolleePaperSubmissionController`